lib/pbench/server/api/resources/endpoint_configure.py

```diff
@@ -0,0 +1,105 @@
+                current ES schema, this is "v5.result-data-sample."
+            result_data_index: The "result-data" index has been broken into
+                "result-data-sample" and "result-data" indices for the
+                Elasticsearch V7 transition. In the current ES schema, this
+                is "v5.result-data."
+            run_index: The "master" run-data index root. In the current ES
+                schema, this is "v6.run-data."
+            run_toc_index: The Elasticsearch V7 index for run TOC data. In
+                the current ES schema, this is "v6.run-toc."
+        api:    A list of the server APIs supported; we give a name, which
+                corresponds to either the current config.json property name
+                (e.g., "elasticsearch") or the current src/service method name
+                being replaced (e.g., "queryControllers").
+            results: Direct access to Pbench data sets for the dashboard; this
+                is currently direct HTTP access through Apache public_html, and
+                not an "API" as such; we will need to adopt this as a real
+                server API when Pbench moves from local filesystem to S3 storage
+                schemas.
+            elasticsearch: The Pbench pass-through URI for the Elasticsearch
+                cluster. This will eventually be superceded by the native
+                Pbench APIs, though it might remain accessible with special
+                user/group privileges to support special cases?
+            graphql: The GraphQL frontend on postgreSQL currently used by the
+                dashboard user mocks. This will be superceded and decprecated
+                by native Pbench user management APIs.
+            queryControllers: Return information about the run documents that
+                were created within a specified range of months.
+            queryMonthIndices: Return the YYYY-mm date strings for all ES
+                vx.run-data.* indices, in descending order.
+
+        TODO: We need an internal mechanism to track the active versions of the
+        various Elasticsearch template documents. We're hardcoding them here and
+        in other APIs. We should consider persisting an equivalent of the
+        mapping table built in "indexer.py" for use across the server APIs.
+
+        Meta TODO: We're giving the pass-through APIs for Elasticsearch and
+        GraphQL here, which implies adoption of Nikhil's and Fuqing's work to
+        move the dashboard away from direct access to the backend DB servers.
+        The alternative would be to expose the direct Elasticsearch and GraphQL
+        URIs here.
+        """
+        try:
+            endpoints = {
+                "metadata": {
+                    "identification": f"Pbench server {self.commit_id}",
+                    "prefix": self.prefix,
+                    "run_index": f"{self.prefix}.v6.run-data.",
```

Comment:
Won't this change require a change in the dashboard for this to work correctly?  And why provide the `prefix` at all?

Reply:
I thought I answered this in another context; but since I'm going through individual file comments now...

Yeah, of course it will. Everything in this PR is designed to provoke a change in the dashboard. In particular, this means that the dashboard no longer needs to know how version and prefix and root index name go together to construct an Elasticsearch index. That sounds like a worthy improvement to me.

Reply:
Right.  And, given that, do we still need line 84 here (since it's incorporated in line 85)?

Reply:
I was being paranoid. But of course you're right. I'll remove it.

Reply:
As long as we still have direct Elasticsearch queries in the dashboard code, it needs to know the index names. We can remove the indices section entirely once we've gotten rid of them all, but there are still a few very messy ones I haven't even gotten to attempting yet.

I suppose we could alternately rejigger the `elasticsearch` endpoint so we could fill in index name templates in the URI rather than the dashboard needing to know; but that's potentially a lot more complicated, and this seemed like a more straightforward replacement of the current static configuration.

Reply:
I am still skeptical as I don't think the current dashboard code works that way, and if you make this change now, it will required more dashboard changes to have it work differently.  See https://github.com/distributed-system-analysis/pbench-dashboard/blob/main/public/endpoints.js#L3.

Reply:
What it means for the dashboard code is that the "root" index name is no longer patched together from the prefix and the hardcoded "base" name; the dashboard gets the whole thing from here and doesn't need to know how to construct it. For now, it still knows how to append the time series suffix; that (along with any need to know anything about the index names) will go away once all of the Elasticsearch queries have been migrated into the server.

The real difference is here, in `moment_constants.js`:

```
    if (index === endpoints.result_index) {
      queryString += `${endpoints.prefix + index + monthValue}-*,`;
    } else {
      queryString += `${endpoints.prefix + index + monthValue},`;
    }
```

vs the new

```
    if (index === endpoints.indices.result_index) {
      queryString += `${index + monthValue}-*,`;
    } else {
      queryString += `${index + monthValue},`;
    }
```

I can see I really need to devote some time to figuring out my e2e mocking problems so I can post a draft dashboard PR with the companion code.